### ISSUE TITLE
Incompatible with Babelify+Browserify?

### ISSUE THREAD

**Issue by wootencl** [ludohenin/gulp-inline-ng2-template#47]
Apologize in advance if this is a question that should be posted to stackoverflow. I'm currently trying to learn angular2 and all its sorrounding frameworks. I thought I would first start out by building a skeleton for angular2 apps. Using the ideas from this [project](https://github.com/shuhei/babel-angular2-app) and the directory structure from this [project](https://github.com/mgechev/angular2-seed) I came up with this [skeleton](https://github.com/wootencl/Angular2_Foundation_Skeleton). The first problem I noticed though is that angular2 has a problem with relative .html templates. After reading this [thread](https://github.com/angular/angular/issues/6053) I stumbled upon your project, which finally gets me to my point. I'm trying to combine the plugin with browserify (based on the example) and chaining that with my babelify transform. Currently I'm getting this error:

```
TypeError: Invalid non-string/buffer chunk while parsing file:...
```

Any help would be much appreciated. I can also post this question to stackoverflow if this is not the right place for it to be. Thanks!

The branch I'm using: https://github.com/wootencl/Angular2_Foundation_Skeleton/tree/ng2_inline_templates

**Comment by ludohenin** [comment]
@wootencl yup, I have to update the `README`, the `browserify` example is out-of-date since from `2.0.0`, the parser accepts a callback. Thta's why, I suppose you're getting this error.

**Comment by wootencl** [comment]
Thanks @ludohenin! That seemed to resolve the issue I was having. For some reason it's still not working on my machine though. It's giving me a `MODULE_NOT_FOUND` error in the `bundle.js` file after compilation. I need to do some further research but was curious if this was something you had maybe run into previously? 

Still using the same [repo](https://github.com/wootencl/Angular2_Foundation_Skeleton/tree/ng2_inline_templates) from original question.

**Comment by wootencl** [comment]
Ended up getting a little free time this weekend and decided to look into it. After taking a glance at your `parser.js` file I noticed that your callback function `done` was returning the results via the second parameter. All I had to do was change my code to reflect that and it worked like a charm. I made a [pull request](https://github.com/ludohenin/gulp-inline-ng2-template/pull/48) fixing the README in the fashion that worked for me. I assumed the first parameter is probably slated for errors. 

Thanks for the awesome plugin!
